contract/src/lib.rs

```diff
@@ -104,216 +130,292 @@ impl TokenFactory {
     #[payable]
     pub fn register(
         &mut self,
-        ft_contract: AccountId, 
+        ft_contract: AccountId,
         deployer_contract: AccountId,
         total_supply: WrappedBalance,
         token_name: String,
         symbol: String,
         icon: Option<String>,
         reference: Option<String>,
         reference_hash: Option<Base64VecU8>,
+        allocations: TokenAllocationInput,
         decimals: u8,
-        
-        initial_release: WrappedBalance,
-        vesting_start_time: WrappedTimestamp,
-        vesting_end_time: WrappedTimestamp,
-        vesting_interval: WrappedDuration,
-        treasury_allocation: WrappedBalance,
     ) {
-        let mut token: State = self.tokens.get(&ft_contract).unwrap_or_default();
         assert!(
             env::attached_deposit() >= 4_000_000_000_000_000_000_000_000,
             "Minimum deposit is 4 NEAR",
         );
+        
+        let mut state_allocations = UnorderedMap::new(b"allocations".to_vec());
```

Comment:
This looks unsafe. If I call register twice, this var will point to the same storage (unless I misunderstand the way Near contract works).